### PR TITLE
LPS-168175 Add descriptive/list view to dm browser; use it in blogs

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-168175

Up until now, the DM browser tag only supported the icon (card) view. Add descriptive and list views for completeness.

Also, this view uses this tag in the Blogs Images section. This change is under feature flag `feature.flag.LPS-168174'.

image

# How do I test this?

Enable the feature flag and go to Blogs > Images. You should see the new UI.

# Things to take into account

The DM tag is still a WIP. There will be some behavior differences between the old Blogs Images interface and the new one (e.g. now it is possible to create folders, which doesn't make much sense in Blogs). These things will be fixed in future PRs.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

**Review by ambrinchaudhary — APPROVED** [review]
LGTM!

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/display/context/RepositoryBrowserTagDisplayContext.java [hunk 1]
```diff
@@ -285,6 +410,91 @@ private BreadcrumbEntry _createBreadcrumbEntry(
...
+					LanguageUtil.get(_httpServletRequest, "rename"));
+			}
+		).add(
+			() -> _fileEntryModelResourcePermission.contains(
+				_themeDisplay.getPermissionChecker(), fileEntry,
+				ActionKeys.DELETE),
+			dropdownItem -> {
+				dropdownItem.putData("action", "delete");
+				dropdownItem.putData(
+					"deleteURL", getDeleteFileEntryURL(fileEntry));
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/display/context/RepositoryBrowserTagDisplayContext.java:434]: The method `getDeleteFileEntryURL` can also be converted to _private_

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
The *rename* action doesn't seem to work on _files_. Could it be we are showing the _fileName_ and changing the _title_ or viceceversa?

**Comment by adolfopa** [comment]
@marcogalluzzi Applied fixes for both of your comments: made method private (squashed with existing commit) and fixed renaming for documents.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1387401_4828

**Review by marcogalluzzi — COMMENTED** [review]
I've tested again merging with last version of master. The *rename* action now works, but I don't know why the *Permissions* action is missing. Maybe these failing tests are also related: 
* [LocalFile.BlogsPermissions#CannotDeleteEntryWhenTeamRoleHasNoPermissions](https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/130522//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_BlogsPermissions_CannotDeleteEntryWhenTeamRoleHasNoPermissions_)
* [LocalFile.BlogsPermissions#CannotAddEntryWhenTeamRoleHasNoPermissions](https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/127136//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_BlogsPermissions_CannotAddEntryWhenTeamRoleHasNoPermissions_)

**Comment by adolfopa** [comment]
The permissions action is not supposed to be there. Those failures must be unrelated, as this is all behind a feature flag.

**Comment by adolfopa** [comment]
ci:test:relevant

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
> The permissions action is not supposed to be there. Those failures must be unrelated, as this is all behind a feature flag.

Sorry, I mixed up this PR with another one

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
ci:reevaluate:1347401_3199

**Pull request closed without merging**